importlib_metadata/_meta.py

```diff
@@ -52,7 +52,7 @@ class SimplePath(Protocol[_T]):
     def joinpath(self, other: Union[str, _T]) -> _T:
         ...  # pragma: no cover
 
-    def __truediv__(self, other: Union[str, _T]) -> _T:
+    def __truediv__(self, other: _T) -> _T:
```

Comment:
I'm not sure this annotation is correct. A SimplePath could accept a `str` as input and return a SimplePath. I understand that `pathlib.Path` is parameterized (the subtype is either `str` or `bytes`), but maybe SimplePath should just assume `str`. The only thing I'm uncertain about is if existing uses might utilize a `Path[bytes]`. That is, maybe we should eliminate `T` and replace it with `SimplePath`... so `__truediv__` accepts `other: Union[str, 'SimplePath']`.